src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -43,16 +43,50 @@ class NotificationActionSerializer(CamelSnakeModelSerializer):
     Django Rest Framework serializer for incoming NotificationAction API payloads
     """
 
-    integration_id = serializers.IntegerField(required=False)
-    sentry_app_id = serializers.IntegerField(required=False)
-    projects = serializers.ListField(child=ProjectField(scope="project:read"), required=False)
-
-    service_type = serializers.CharField()
-    target_type = serializers.CharField()
-    trigger_type = serializers.CharField()
-
-    target_identifier = serializers.CharField(required=False)
-    target_display = serializers.CharField(required=False)
+    integration_id = serializers.IntegerField(
+        help_text="""ID of integration used as the notification service. For example, ID of
+            Slack integration if Slack is used to notify org members.
+            This field is required if **service_type** is `slack`, `pagerduty` or `opsgenie`.""",
+        required=False,
+    )
+
+    # Optional and not needed for spike protection so not documenting
+    sentry_app_id = serializers.IntegerField(
+        required=False,
+    )
+    projects = serializers.ListField(
+        help_text="""List of projects slugs that Notificaton Action is created for""",
+        child=ProjectField(scope="project:read"),
+        required=False,
+    )
+
+    service_type = serializers.CharField(
+        help_text="Service that is used for sending the notification\n"
+        + """- `email`\n"""
+        + """- `slack`\n"""
+        + """- `pagerduty`\n"""
+        + """- `opsgenie`\n"""
+        + """- `sentry_notification`\n"""
+    )
+    target_type = serializers.CharField(
+        help_text="""Type of notification recipient\n"""
+        + """ - `specific`\n"""
```

Comment:
I don't know what `specific` or `sentry_app` is here. Should each bullet point have a quick description or example?

Reply:
@cathteng can you provide more context here. The example in product is always "specific". I think I should not include sentry_app because we decided to hide sentry_app_id and was not related to spike protection.

Reply:
`specific` is the only one used for Notification Actions so it might make more sense to only show that option for this field. It comes from [this class](https://github.com/getsentry/sentry/blob/master/src/sentry/models/notificationaction.py#L100) and [this enum](https://github.com/getsentry/sentry/blob/master/src/sentry/models/notificationaction.py#L76) since it's an abstraction of the metric alerts 'AlertRuleTriggerAction' model. That skeleton is shared and so this model allows for `user`, `team` and `sentry_app` but they aren't used.

Those values are used in the code along with `target_identifier` to figure out how to send a notification. So if `target_type` is `user` then the `target_identifier` will be the User's ID. A `target_type` of `specific`, pretty much just means, we don't have a database identifer to point to, this identifier is specific to the service sending the notification (e.g. a pagerduty service ID, or a slack channel ID)

Reply:
@leeandher so I only show specific. For slack integration, my understanding is that `target_identier`
 is the slack channel ID no? so:
 ```
integration_id = SLACK_INTEGRATION_ID
service_type = slack
target_type = specific
target_identifier = channel_id
target_display = channel_name
```

Reply:
yep thats exactly correct -- the `user`, `team` and `sentry_app_id` values are used in metric alerts, not notification actions.  The `target_type` just hints how to interpret the identifier, you are correct that for slack, the identifier is the slack channel id. 

They share a base abstract class which is why they appear in the serializer, but no one's implemented any notif actions that will use those values, so it's fine just saying 'specific' in the docs.